### PR TITLE
Allow passing NULL for error (fs and btrfs plugins)

### PR DESCRIPTION
**Pull request by vojtechtrefny**
One step closer to fixing #527

### CODE REVIEW

**Review by tbzatek — COMMENTED** [review]
#676 ... sorry

**Comment by vojtechtrefny** [comment]
> #676 ... sorry

I know. That will be the last step, mass replacing `(out) (allow-none)` with `(optional)` :-)

File: src/plugins/fs/xfs.c [hunk 1]
```diff
@@ -203,15 +203,20 @@ gboolean bd_fs_xfs_wipe (const gchar *device, GError **error) {
...
         return FALSE;
 
-    ret = bd_utils_exec_and_report_error (args, NULL, error);
-    if (!ret && *error &&  g_error_matches ((*error), BD_UTILS_EXEC_ERROR, BD_UTILS_EXEC_ERROR_FAILED))
-        /* non-zero exit status -> the fs is not clean, but not an error */
-        /* TODO: should we check that the device exists and contains an XFS FS beforehand? */
-        g_clear_error (error);
+    ret = bd_utils_exec_and_report_error (args, NULL, &l_error);
+    if (!ret) {
+        if (l_error &&  g_error_matches (l_error, BD_UTILS_EXEC_ERROR, BD_UTILS_EXEC_ERROR_FAILED)) {
```
Comment (tbzatek) [inline src/plugins/fs/xfs.c:213]: extra space! :-)

**Review by tbzatek — APPROVED** [review]
Whoa, what a large commit. Thanks for doing this!

**Pull request merged**